### ISSUE TITLE
Abruptly closed connection causes bun install and fetch to crash

### ISSUE THREAD

**Issue by ziyadev** [oven-sh/bun#12546]
### How can we reproduce the crash?

I'm encountering issues while using the official Dockerfile configuration to build my NestJS app. The build process stalls at the following step:
`RUN cd /temp/dev && bun install --frozen-lockfile`

### Relevant log output

```shell
Step 5/21 : RUN cd /temp/dev && bun install --frozen-lockfile
 ---> Running in 8042b45383ff
bun install v1.1.18 (5a0b9352)
Resolving dependencies
Resolved, downloaded and extracted [2]
============================================================
Bun v1.1.18 (5a0b9352) Linux x64 (baseline)
Linux Kernel v6.8.11 | glibc v2.31
Args: "bun" "install" "--frozen-lockfile"
Features: extracted_packages(189)
Elapsed: 384584ms | User: 1636ms | Sys: 2961ms
RSS: 3.76GB | Peak: 0.42GB | Commit: 3.76GB | Faults: 200

panic: Segmentation fault at address 0xB0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
Illegal instruction (core dumped)
The command '/bin/sh -c cd /temp/dev && bun install --frozen-lockfile' returned a non-zero code: 132
```

### Stack Trace (bun.report)

Bun v1.1.18 ([`5a0b935`](<https://github.com/oven-sh/bun/tree/5a0b93523134ca0043ee7ae84f01003aa4b7ff69>)) on linux x86_64_baseline [InstallCommand]

Segmentation fault at address 0x000000B0

- `??`
- [`http.zig:2999`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/http.zig#L2999>): `src.http.closeAndFail__anon_149944`
- [`http.zig:0`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/http.zig#L0>): `src.http.onData__anon_149951`
- [`http.zig:580`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/http.zig#L580>): `src.deps.uws.NewSocketHandler`
- [`openssl.c:526`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/crypto/openssl.c#L526>): `ssl_on_data`
- [`loop.c:388`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/loop.c#L388>): `us_loop_run_bun_tick`
- [`uws.zig:1283`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/deps/uws.zig#L1283>): `src.http.HTTPThread.processEvents`
- [`http.zig:814`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/http.zig#L814>): `Thread.PosixThreadImpl.spawn__anon_136392.Instance.entryFn`
- `??`

**Comment by yooouuri** [comment]
Same over here

```
FROM oven/bun:1 AS base
WORKDIR /usr/src/app

FROM base AS install
RUN mkdir -p /temp/dev
COPY package.json bun.lockb /temp/dev/
RUN cd /temp/dev && bun install --frozen-lockfile

RUN mkdir -p /temp/prod
COPY package.json bun.lockb /temp/prod/
RUN cd /temp/prod && bun install --frozen-lockfile --production

FROM base AS prerelease
COPY --from=install /temp/dev/node_modules node_modules
COPY . .
ENV NODE_ENV=production
RUN bun run --bun build

FROM base AS release
COPY --from=install /temp/prod/node_modules node_modules
COPY --from=prerelease /usr/src/app/build/server ./build/server
COPY --from=prerelease /usr/src/app/build/client ./build/client
COPY --from=prerelease /usr/src/app/package.json .

USER bun
EXPOSE 3000/tcp
ENTRYPOINT [ "bun", "run", "start" ]
```

```
Step 6/22 : RUN cd /temp/dev && bun install --frozen-lockfile
 ---> Running in 4fc091e27a5e
bun install v1.1.29 (6d43b366)
Resolving dependencies
Resolved, downloaded and extracted [2]
```

**Comment by farhan-helmy** [comment]
same here

**Comment by arsu4ka** [comment]
anybody got a fix? stuck with the same issue, can't build the docker image

**Comment by suryababus** [comment]
Stuck on the same issue.

**Comment by cirospaciari** [comment]
This Issue was caused by not properly marking the socket closed when the socket was abrupt closed. This is current fixed in v1.1.39. If you encounter any similar Issue please open a new Issue with a reproduction.

**Comment by SupeManDeveloper** [comment]
![Image](https://github.com/user-attachments/assets/b6678054-a35e-4078-b5e4-16c57a4d3aeb)
Latest version (v1.1.39) having some issues. Use v1.1.38 instead.
